wsmaster/codenvy-hosted-api-audit/src/main/java/com/codenvy/api/audit/server/AuditService.java

```diff
@@ -0,0 +1,194 @@
+import org.eclipse.che.api.user.server.model.impl.UserImpl;
+import org.eclipse.che.api.workspace.server.WorkspaceManager;
+import org.eclipse.che.api.workspace.server.model.impl.WorkspaceImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.StreamingOutput;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.text.SimpleDateFormat;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Defines Audit report REST API.
+ *
+ * @author Igor Vinokur
+ */
+@Path("/audit")
+public class AuditService extends Service {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AuditService.class);
+
+    private final AdminUserDao          adminUserDao;
+    private final WorkspaceManager      workspaceManager;
+    private final PermissionManager     permissionManager;
+    private final CodenvyLicenseManager licenseManager;
+
+    @Inject
+    public AuditService(AdminUserDao adminUserDao,
+                        WorkspaceManager workspaceManager,
+                        PermissionManager permissionManager,
+                        CodenvyLicenseManager licenseManager) {
+        this.adminUserDao = adminUserDao;
+        this.workspaceManager = workspaceManager;
+        this.permissionManager = permissionManager;
+        this.licenseManager = licenseManager;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public StreamingOutput getReport() throws ServerException {
+        File file = initializeFileInTempDirectory();
+
+        printHeader(file);
+        printAllUsersInfo(file);
+
+        return initializeStreamAndCleanTempDirectory(file);
+    }
+
+    private File initializeFileInTempDirectory() {
+        final File tempDir = Files.createTempDir();
+        return new File(tempDir, "report.txt");
+    }
+
+    private void printHeader(File file) throws ServerException {
+        //TODO: rework to getTotalCount() method from JpaUserDao, when it will be merged to master.
+        appendToFile("Number of all users: " + adminUserDao.getAll(1, 0).getTotalItemsCount() + "\n", file);
+        final CodenvyLicense license;
+        try {
+            license = licenseManager.load();
+            appendToFile("Number of users licensed: " + license.getNumberOfUsers() + "\n", file);
+            appendToFile(
+                    "Date when license expires: " + new SimpleDateFormat("dd MMMM yyyy").format(license.getExpirationDate()) + "\n", file);
+        } catch (LicenseException e) {
+            appendToFile("Failed to retrieve license!\n", file);
+            LOG.error(e.getMessage(), e);
+        }
+    }
+
+    private void printAllUsersInfo(File file) throws ServerException {
+        int skipItems = 0;
+        while (true) {
```

Comment:
But you can get itemsBefore aka skipCount and pageSize aka maxItems for next page